plugins/rts54hub/fu-rts54hub-rtd21xx-mergeinfo.c

```diff
@@ -0,0 +1,481 @@
+					   (FuDeviceLockerFunc)fu_rts54hub_rtd21xx_mergeinfo_exit,
+					   error);
+	if (locker == NULL)
+		return FALSE;
+	if (!fu_rts54hub_rtd21xx_mergeinfo_ensure_version(self, error))
+		return FALSE;
+
+	/* success */
+	return TRUE;
+}
+
+static gboolean
+fu_rts54hub_rtd21xx_mergeinfo_reload(FuDevice *device, GError **error)
+{
+	FuRts54HubDevice *parent = FU_RTS54HUB_DEVICE(fu_device_get_parent(device));
+	g_autoptr(FuDeviceLocker) locker = NULL;
+
+	/* open parent device */
+	locker = fu_device_locker_new(parent, error);
+	if (locker == NULL)
+		return FALSE;
+	return fu_rts54hub_rtd21xx_mergeinfo_setup(device, error);
+}
+
+static gboolean
+fu_rts54hub_rtd21xx_mergeinfo_write_firmware(FuDevice *device,
+					     FuFirmware *firmware,
+					     FuProgress *progress,
+					     FwupdInstallFlags flags,
+					     GError **error)
+{
+	FuRts54hubRtd21xxMergeinfo *self = FU_RTS54HUB_RTD21XX_MERGEINFO(device);
+	guint8 read_buf[VERSION_NUMBER_COUNT] = {0x0};
+	guint8 merge_version[VERSION_NUMBER_COUNT] = {0x00};
+	g_autoptr(FuDeviceLocker) locker = NULL;
+	g_autoptr(GInputStream) stream = NULL;
+
+	/* progress */
+	fu_progress_set_id(progress, G_STRLOC);
+	fu_progress_add_flag(progress, FU_PROGRESS_FLAG_GUESSED);
+	fu_progress_add_step(progress, FWUPD_STATUS_DEVICE_WRITE, 40, "write");
+	fu_progress_add_step(progress, FWUPD_STATUS_DEVICE_READ, 50, "read");
+	fu_progress_add_step(progress, FWUPD_STATUS_DEVICE_BUSY, 10, "finish");
+
+	/* open device */
+	locker = fu_device_locker_new(self, error);
+	if (locker == NULL)
+		return FALSE;
+
+	/* simple image */
+	stream = fu_firmware_get_stream(firmware, error);
+	if (stream == NULL)
+		return FALSE;
+
+	/* get merge version */
+	if (!fu_input_stream_read_safe(stream,
+				       merge_version,
+				       sizeof(merge_version),
+				       0, /* dst */
+				       0, /* src */
+				       VERSION_NUMBER_COUNT,
+				       error)) {
+		g_prefix_error(error, "failed to get merge version info: ");
+		return FALSE;
+	}
+
+	/* write version*/
+	if (!fu_rts54hub_rtd21xx_mergeinfo_write_version(self,
+							 merge_version,
+							 VERSION_NUMBER_COUNT,
+							 error)) {
+		g_prefix_error(error, "failed to write merge version: ");
+	}
+
+	/* wait for device ready */
+	fu_device_sleep(FU_DEVICE(self), 1000); /* ms */
+
+	fu_progress_step_done(progress);
+
+	if (!fu_rts54hub_rtd21xx_mergeinfo_read_version(self, read_buf, error)) {
+		g_prefix_error(error, "failed to read merge version: ");
+		return FALSE;
+	}
+
+	fu_progress_step_done(progress);
+
+	if (!fu_memcmp_safe(read_buf,
+			    VERSION_NUMBER_COUNT,
+			    0x0,
+			    merge_version,
+			    VERSION_NUMBER_COUNT,
+			    0x0,
+			    VERSION_NUMBER_COUNT,
+			    error)) {
+		g_prefix_error(error, "failed to compare merge version: ");
+		return FALSE;
+	}
+
+	fu_progress_step_done(progress);
+
+	/* success */
+	return TRUE;
+}
+
+static void
+fu_rts54hub_rtd21xx_mergeinfo_init(FuRts54hubRtd21xxMergeinfo *self)
+{
+}
```

Comment:
I'm suprised not to see UPDATABLE -- is the mergeinfo just for *display*? I thought you were installing the version onto it...